tidy3d/plugins/dispersion/fit_fast.py

```diff
@@ -0,0 +1,668 @@
+        min_imag = np.zeros(len(passivity_omega))
+        min_imag[PASSIVITY_NUM:] = min_loss
+        if not np.all(np.imag(self.evaluate(passivity_omega)) > min_imag):
+            return False
+        try:
+            _, vals = self.imag_extrema
+            return np.all(np.imag(vals) > 0)
+        except np.linalg.LinAlgError:
+            return False
+
+    def get_error(self, omega, eps, weights) -> ArrayFloat1D:
+        """Get RMS error"""
+        diff = self.evaluate(omega) - eps
+        if weights is None:
+            square = np.abs(diff) ** 2
+        else:
+            square = (weights[0][::-1] * np.real(diff)) ** 2 + (
+                weights[1][::-1] * np.imag(diff)
+            ) ** 2
+        return np.sqrt(np.sum(square) / len(eps))
+
+
+class FastDispersionFitter(DispersionFitter):
+    """Tool for fitting refractive index data to get a
+    dispersive medium described by :class:`.PoleResidue` model."""
+
+    eps_inf: Optional[float] = Field(
+        1,
+        title="eps_inf",
+        description="Value of eps_inf to use for fit. If None, then fit eps_inf as well.",
+    )
+
+    weights: Union[Tuple[float, float], Tuple[ArrayFloat1D, ArrayFloat1D]] = Field(
+        None,
+        title="Weights",
+        description="(real weight, imaginary weight), optionally as a function of wavelength. "
+        "Used to weight the rms error objective function.",
+    )
+
+    min_loss_wvl: ArrayFloat1D = Field(
+        [], title="Min loss wavelengths", description="Wavelengths to impose min loss condition. "
+    )
+
+    min_loss: Union[float, ArrayFloat1D] = Field(
+        0, title="Min loss", description="Min loss at min_loss_freqs"
+    )
+
+    passivity_num_iters: int = Field(
+        1,
+        title="Number of iterations for passivity",
+        description="Number of iterations for passivity (and min loss) enforcement.",
+    )
+
+    @validator("weights", always=True)
+    def _weights_len(cls, val, values):
+        """Weights must be correct length."""
+        if val is None:
+            return val
+        wvl = values["wvl_um"]
+        if isinstance(val[0], float):
+            return (val[0] * np.ones(len(wvl)), val[1] * np.ones(len(wvl)))
+        if not len(val[0]) == len(val[1]) == len(wvl):
+            raise ValidationError("Fitting weights must be same length as fitting data.")
+        return val
+
+    @validator("min_loss", always=True)
+    def _min_loss_len(cls, val, values):
+        """Weights must be correct length."""
+        wvl = values["min_loss_wvl"]
+        if isinstance(val, float):
+            return val * np.ones(len(wvl))
+        if not len(val) == len(wvl):
+            raise ValidationError("The min_loss must be same length as min_loss_wvl.")
+        return val
+
+    @cached_property
+    def _min_loss_omegas(self) -> ArrayFloat1D:
+        return 2 * np.pi * HBAR * C_0 / np.array(self.min_loss_wvl[::-1])
+
+    def fit_fast(  # pylint: disable=too-many-locals
```

Comment:
Because it inherits from DispersionFitter, calling `FastDispersionFitter.fit` actually uses the base class. Isn't it better to override `fit` instead of creating `fit_fast`? It can be confusing as it is right now.